CommunityToolkit.Mvvm.SourceGenerators/ComponentModel/ObservableRecipientGenerator.cs

```diff
@@ -117,19 +117,18 @@ protected override ImmutableArray<MemberDeclarationSyntax> FilterDeclaredMembers
         // If the target type has no constructors, generate constructors as well
         if (!info.HasExplicitConstructors)
         {
-            foreach (ConstructorDeclarationSyntax ctor in memberDeclarations.OfType<ConstructorDeclarationSyntax>())
+            foreach (ConstructorDeclarationSyntax originalConstructor in memberDeclarations.OfType<ConstructorDeclarationSyntax>())
             {
-                string text = ctor.NormalizeWhitespace().ToFullString();
-                string replaced = text.Replace("ObservableRecipient", info.TypeName);
+                ConstructorDeclarationSyntax modifiedConstructor = originalConstructor.WithIdentifier(Identifier(info.TypeName));
 
                 // Adjust the visibility of the constructors based on whether the target type is abstract.
                 // If that is not the case, the constructors have to be declared as public and not protected.
                 if (!info.IsAbstract)
                 {
-                    replaced = replaced.Replace("protected", "public");
+                    modifiedConstructor = modifiedConstructor.WithModifiers(TokenList(Token(SyntaxKind.PublicKeyword)));
```

Comment:
What if the constructor has other modifiers as well, like `unsafe`? This would remove all of them. Or is this constructor not from user code but generated by the generator and it's guaranteed that it won't have other modifiers?

Reply:
If preserving other modifiers is needed, try something like this:
```c#
modifiedConstructor = modifiedConstructor.WithModifiers(
    modifiedConstructor.Modifiers.Replace(
        modifiedConstructor.Modifiers.First(token => token.Kind() is SyntaxKind.ProtectedKeyword),
        SyntaxFactory.Token(SyntaxKind.PublicKeyword)));
```

Reply:
The syntax tree is from the `ObservableRecipient.cs` embedded resource file, so yeah we can assume that `protected` is the only modifier that could possibly be there. These trees are not provided by code parsed from consumers 🙂